pkg/controller/command/command_upgrade.go

```diff
@@ -30,8 +49,156 @@ func performUpgradeStepIfNeeded(commandCR *contrail.Command, currentDeployment *
 		currentDeployment.Spec.Replicas = commandCR.Spec.CommonConfiguration.Replicas
 		commandCR.Status.UpgradeState = contrail.CommandNotUpgrading
 	}
+	return nil
+}
+
+func (r *ReconcileCommand) deleteMigrationJob(commandCR *contrail.Command) error {
+	job := &batch.Job{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      commandCR.Name + "-upgrade-job",
+			Namespace: commandCR.Namespace,
+		},
+	}
+	pp := meta.DeletePropagationForeground
+	err := r.client.Delete(context.Background(), job, &client.DeleteOptions{PropagationPolicy: &pp})
+	if !errors.IsNotFound(err) {
+		return err
+	}
+	return nil
 }
 
+func getOldCommandImage(list []core.Container, name string) string {
```

Comment:
This function only iterates over list of containers and look for the container image. It can be used for any image not only "old command"?

I prefer 'containers' over 'list' as argument

Reply:
ok I can change that